### DISCUSSION TITLE
How To Use Jest With Parcel 2 Imports

### DISCUSSION THREAD

**Discussion by tsmith123** [parcel-bundler/parcel#5217] [Q&A] [answered]
I'm attempting to migrate to v2 of Parcel having used v1 for a year now. The migration docs (https://v2.parceljs.org/getting-started/migration/) say that when importing non-code assets then this:

`import Icon from '@assets/images/icon.svg'` (I use babel module resolver for aliases)

becomes:

`import Icon from 'url:../../assets/images/icon.svg'`

Also, my tests now break because `jest` will not recognize the paths using `url:`

`Cannot find module 'url:../../assets/images/icon.svg'`

**Question 1:** Is there a way to use `url:` with existing alisases?
**Question 2:** How can I get Jest to work with the recommended change?

**Comment by mischnic** [comment, marked as answer]
> Is there a way to use url: with existing alisases?

The specifier after `url:` is resolved normally. Does https://v2.parceljs.org/features/module-resolution#aliases work for your?

> How can I get Jest to work with the recommended change?

Maybe https://jestjs.io/docs/en/configuration#modulenamemapper-objectstring-string--arraystring?
This would strip the `url:` part:

```json
  "moduleNameMapper": {
    "^url:(.*)+$": "$1",
  }
```
You might want to map this to some stub file instead
